pkg/migrate/storage.go

```diff
@@ -92,43 +92,57 @@ func GetMigrationLogs() string {
 	return migrationLogs
 }
 
-// IsMigrationReady returns true if the cluster is ready to migrate storage
-// this is true if Ceph is setup/healthy, the ceph storageclass is present in the cluster, and the ceph object store user exists
-func IsMigrationReady(ctx context.Context, config ekcoops.Config, controllers types.ControllerConfig) (bool, string, error) {
+// MigrationReadyStatus represents the status of the migration readiness check, includes
+// a reason and also the number of nodes in the cluster.
+type MigrationReadyResult struct {
+	Ready   bool   `json:"ready"`
+	Reason  string `json:"reason"`
+	NrNodes int    `json:"nrNodes"`
```

Comment:
Could we also return the required number of nodes? RookMinimumNodeCount won't _always_ be 3